src/stream/stream/mod.rs

```diff
@@ -750,6 +752,41 @@ extension_trait! {
             FoldFuture::new(self, init, f)
         }
 
+        #[doc = r#"
+            Call a closure on each element of the stream.
+
+            # Examples
+
+            ```
+            # fn main() { async_std::task::block_on(async {
+            #
+            use async_std::prelude::*;
+            use std::collections::VecDeque;
+            use std::sync::mpsc::channel;
+
+            let (tx, rx) = channel();
+
+            let s: VecDeque<usize> = vec![1, 2, 3].into_iter().collect();
+            let sum = s.for_each(move |x| tx.clone().send(x).unwrap()).await;
```

Comment:
Could this method maybe take an async closure as well, similar to #262? Thanks!

Reply:
Yup, like i said i intend to update #262 as soon as this is merged. Or we could do this the other way around: merge #262 and update this.

Reply:
oh I missed that -- yeah let's go ahead and merge this first!